### ISSUE TITLE
Unable to lauch app from Xcode on simulator on dev channel

### ISSUE THREAD

**Issue by blaugold** [flutter/flutter#78418]
## Steps to Reproduce

1. Run the following commands:

```bash
flutter channel dev
flutter upgrade

flutter create launch_from_xcode_issue
cd launch_from_xcode_issue

flutter build ios

open ios/Runner.xcworkspace
```

2. Launch the app from Xcode on a simulator

**Expected results:** App launches as normal.

**Actual results:** App crashes: `EXC_BAD_ACCESS (code=50, address=0x104357000)`
```
dyld_sim`memcmp:
    0x104123231 <+0>:  pushq  %rbp
    0x104123232 <+1>:  movq   %rsp, %rbp
    0x104123235 <+4>:  testq  %rdx, %rdx
    0x104123238 <+7>:  je     0x104123252               ; <+33>
    0x10412323a <+9>:  xorl   %ecx, %ecx
->  0x10412323c <+11>: movzbl (%rdi,%rcx), %eax
    0x104123240 <+15>: movzbl (%rsi,%rcx), %r8d
    0x104123245 <+20>: cmpb   %r8b, %al
    0x104123248 <+23>: jne    0x104123256               ; <+37>
    0x10412324a <+25>: incq   %rcx
    0x10412324d <+28>: cmpq   %rcx, %rdx
    0x104123250 <+31>: jne    0x10412323c               ; <+11>
    0x104123252 <+33>: xorl   %eax, %eax
    0x104123254 <+35>: jmp    0x104123259               ; <+40>
    0x104123256 <+37>: subl   %r8d, %eax
    0x104123259 <+40>: popq   %rbp
    0x10412325a <+41>: retq   
```
```
dyld`_dyld_start:
    0x10732f000 <+0>:  popq   %rdi
    0x10732f001 <+1>:  pushq  $0x0
    0x10732f003 <+3>:  movq   %rsp, %rbp
    0x10732f006 <+6>:  andq   $-0x10, %rsp
    0x10732f00a <+10>: subq   $0x10, %rsp
    0x10732f00e <+14>: movl   0x8(%rbp), %esi
    0x10732f011 <+17>: leaq   0x10(%rbp), %rdx
    0x10732f015 <+21>: leaq   -0x101c(%rip), %rcx
    0x10732f01c <+28>: leaq   -0x8(%rbp), %r8
    0x10732f020 <+32>: callq  0x10732f062               ; dyldbootstrap::start(dyld3::MachOLoaded const*, int, char const**, dyld3::MachOLoaded const*, unsigned long*)
->  0x10732f025 <+37>: movq   -0x8(%rbp), %rdi
    0x10732f029 <+41>: cmpq   $0x0, %rdi
    0x10732f02d <+45>: jne    0x10732f03f               ; <+63>
    0x10732f02f <+47>: movq   %rbp, %rsp
    0x10732f032 <+50>: addq   $0x8, %rsp
    0x10732f036 <+54>: movq   $0x0, %rbp
    0x10732f03d <+61>: jmpq   *%rax
    0x10732f03f <+63>: addq   $0x10, %rsp
    0x10732f043 <+67>: pushq  %rdi
    0x10732f044 <+68>: movq   0x8(%rbp), %rdi
    0x10732f048 <+72>: leaq   0x10(%rbp), %rsi
    0x10732f04c <+76>: leaq   0x8(%rsi,%rdi,8), %rdx
    0x10732f051 <+81>: movq   %rdx, %rcx
    0x10732f054 <+84>: movq   (%rcx), %r8
    0x10732f057 <+87>: addq   $0x8, %rcx
    0x10732f05b <+91>: testq  %r8, %r8
    0x10732f05e <+94>: jne    0x10732f054               ; <+84>
    0x10732f060 <+96>: jmpq   *%rax
```

<details>
  <summary>Logs</summary>

```
Analyzing flutter_ios_sim_issue...                                      
No issues found! (ran in 2.5s)
```

```
[✓] Flutter (Channel dev, 2.1.0-12.1.pre, on macOS 11.2.3 20D91 darwin-x64, locale en-DE)
    • Flutter version 2.1.0-12.1.pre at /Users/gabriel/lib/flutter
    • Framework revision 8264cb3e8a (7 days ago), 2021-03-10 12:37:57 -0800
    • Engine revision 711ab3fda0
    • Dart version 2.13.0 (build 2.13.0-116.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/gabriel/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/gabriel/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/gabriel/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • DC5FB20C-5DF0-42A4-B4A0-F91ECA27ED94 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64     • macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 89.0.4389.90

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
Thanks @blaugold for reporting this. I was able to replicate this crash, but only on dev. On latest master, it works properly without any issues.

Dev:
```

dyld_sim`memcmp:
    0x103995231 <+0>:  pushq  %rbp
    0x103995232 <+1>:  movq   %rsp, %rbp
    0x103995235 <+4>:  testq  %rdx, %rdx
    0x103995238 <+7>:  je     0x103995252               ; <+33>
    0x10399523a <+9>:  xorl   %ecx, %ecx
->  0x10399523c <+11>: movzbl (%rdi,%rcx), %eax
    0x103995240 <+15>: movzbl (%rsi,%rcx), %r8d
    0x103995245 <+20>: cmpb   %r8b, %al
    0x103995248 <+23>: jne    0x103995256               ; <+37>
    0x10399524a <+25>: incq   %rcx
    0x10399524d <+28>: cmpq   %rcx, %rdx
    0x103995250 <+31>: jne    0x10399523c               ; <+11>
    0x103995252 <+33>: xorl   %eax, %eax
    0x103995254 <+35>: jmp    0x103995259               ; <+40>
    0x103995256 <+37>: subl   %r8d, %eax
    0x103995259 <+40>: popq   %rbp
    0x10399525a <+41>: retq   

```

Master:

<img width="366" alt="Screenshot 2021-03-17 at 7 43 02 PM" src="https://user-images.githubusercontent.com/67046386/111481620-2656e480-8759-11eb-929f-840cfc0e77be.png">

<details>
<summary>latest dev and master flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, 2.1.0-12.1.pre, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.1.0-12.1.pre at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 8264cb3e8a (7 days ago), 2021-03-10 12:37:57 -0800
    • Engine revision 711ab3fda0
    • Dart version 2.13.0 (build 2.13.0-116.0.dev)

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 89.0.4389.82

! Doctor found issues in 1 category.


[✓] Flutter (Channel master, 2.1.0-13.0.pre.133, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.1.0-13.0.pre.133 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 4c0a8e01c1 (12 hours ago), 2021-03-16 18:43:03 -0700
    • Engine revision 525271dd80
    • Dart version 2.13.0 (build 2.13.0-143.0.dev)

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 89.0.4389.82

! Doctor found issues in 1 category.

```

</details>

**Comment by jmagman** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/77397 and was resolved in https://skia-review.googlesource.com/c/skia/+/380716 via https://github.com/flutter/flutter/pull/77795.

We thought this was fixed in 2.1, we'll investigate hotfixing this https://github.com/flutter/flutter/issues/77397#issuecomment-810655013.
